httpcore/_async/http2.py

```diff
@@ -51,7 +51,7 @@ def __init__(
         self._exhausted_available_stream_ids = False
 
     def __repr__(self) -> str:
-        return f"<AsyncHTTP2Connection [{self._state}]>"
+        return f"<AsyncHTTP2Connection [{self._state.name}]>"
```

Comment:
```DeprecationWarning: in 3.12 format() will use the enum member, not the enum member's value;
use a format specifier, such as :d for an IntEnum member, to maintain the current display.
```

Reply:
https://www.python.org/dev/peps/pep-0663/ that's this right?

Reply:
I think we should change the AsyncHTTP2Connection to use the repr of its state

```suggestion
        return f"<AsyncHTTP2Connection [{self._state!r}]>"
```

Reply:
`Enum.__repr__` has different behavior between python3.10+ and below, we'll get different output among python versions.

Reply:
That's working as designed imho